includes/class-wcpdf-main.php

```diff
@@ -846,36 +850,78 @@ public function set_header_logo_height( $document_type, $document = null ) {
 		}
 	}
 
+	public function schedule_temporary_files_cleanup() {
+		$cleanup_age_days = isset( WPO_WCPDF()->settings->debug_settings['cleanup_days'] ) ? floatval( WPO_WCPDF()->settings->debug_settings['cleanup_days'] ) : 7.0;
+		$delete_timestamp = time() - ( intval ( DAY_IN_SECONDS * $cleanup_age_days ) );
+		$this->temporary_files_cleanup( $delete_timestamp );
+	}
+	
 	/**
-	 * Remove attachments older than 1 week (daily, hooked into wp_scheduled_delete )
+	 * Remove temporary files from paths older than 1 week (daily, hooked into wp_scheduled_delete )
+	 * @param string|int  $delete_timestamp timestamp
+	 * 
+	 * @return array Output message
 	 */
-	public function attachments_cleanup() {
-		if ( !function_exists("glob") || !function_exists('filemtime') ) {
-			// glob is required
-			return;
-		}
+	public function temporary_files_cleanup( $delete_timestamp = '' ) {
+		global $wp_filesystem;
 
-		
-		if ( !isset( WPO_WCPDF()->settings->debug_settings['enable_cleanup'] ) ) {
-			return;
+		if ( ! function_exists( 'WP_Filesystem' ) ) {
+			require_once( ABSPATH . 'wp-admin/includes/file.php' );
 		}
 
+		if ( ! WP_Filesystem() || is_null( WP_Filesystem() ) ) {
+			/* translators: WP_Filesystem */
+			return array( 'error' => sprintf( esc_html__( "%s couldn't be initiated!", 'woocommerce-pdf-invoices-packing-slips' ), 'WP_Filesystem' ) );
+		}
 
-		$cleanup_age_days = isset(WPO_WCPDF()->settings->debug_settings['cleanup_days']) ? floatval(WPO_WCPDF()->settings->debug_settings['cleanup_days']) : 7.0;
-		$delete_timestamp = time() - ( intval ( DAY_IN_SECONDS * $cleanup_age_days ) );
-
-		$tmp_path = $this->get_tmp_path('attachments');
+		$paths_to_cleanup    = apply_filters( 'wpo_wcpdf_cleanup_tmp_paths', array(
+			$this->get_tmp_path( 'attachments' ),
+			$this->get_tmp_path( 'dompdf' ),
+		) );
+		$exclude_directories = apply_filters( 'wpo_wcpdf_cleanup_exclude_directories', array() );
+		$allowed_filetypes   = apply_filters( 'wpo_wcpdf_cleanup_allowed_filetypes', array( 'pdf' ) );
```

Comment:
This seems directory specific to me. In the `attachments` folder we only want to clean the `pdf`, but in the `dompdf` folder we want to delete everything, most files there don't even have an extension I believe.

Isn't it easier to just clean everything except `.htaccess` and `index.php` ?